types/module/server/derived_module_key.go

```diff
@@ -0,0 +1,49 @@
+package server
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type DerivedModuleKey struct {
```

Comment:
this shouldn't be moved to cosmos-sdk, this is for ADR-033 which shouldn't be added yet